### PR TITLE
add item to cart and intercept

### LINKED ISSUE #72: E2E > Add item to cart
https://magento.softwaretestingboard.com/

GIVEN I am on home page
WHEN I open MEN dropdown from navbar
AND I choose TOPS > TEES
THEN I am redirected to `/men/tops-men/tees-men.html`

GIVEN I am on tops tees page
WHEN I `filter` by `color green`
THEN Four items are displayed

WHEN I click on `Atomic Endurance` T-shirt item
THEN I am redirected to `/atomic-endurance-running-tee-v-neck.html`

GIVEN I am at `/atomic-endurance-running-tee-v-neck.html`
WHEN I choose size XL
AND I choose color green
AND I choose quantity 3
AND I click `Add to cart`
THEN a get request is called at `/customer/section/load/` end point
AND `product_id` with value `494` is present in the `cart.items` property

### PR DESCRIPTION
**Pull request by pmirel**
https://github.com/adrianmaciuc/break-it-playwright-javascript-project/issues/72

closes #72

### CODE REVIEW

**Review by RaduMoldovan30 — DISMISSED** [review]

File: tests/cart/addItemToCart.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  await page
+    .getByAltText('Atomic Endurance Running Tee (V-neck)-XS-Green')
+    .click();
+
+  await expect(page.url()).toEqual(
+    'https://magento.softwaretestingboard.com/atomic-endurance-running-tee-v-neck.html'
+  );
+});
+
+test('Add item to cart', async ({ page }) => {
```
Comment (adrianmaciuc) [inline tests/cart/addItemToCart.spec.js:31]: merge this with the above, have just one single test

File: tests/homePage/homePageHeader2.spec.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ test('Search suggestion in header', async ({ page }) => {
 test('Logo click redirect home page', async ({ page }) => {
```
Comment (adrianmaciuc) [inline tests/homePage/homePageHeader2.spec.js:29]: mix with above, have just one single e2e

File: tests/cart/addItemToCart.spec.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  await page.goto(
+    'https://magento.softwaretestingboard.com/atomic-endurance-running-tee-v-neck.html'
+  );
+  await page.getByLabel('XL').click();
+  await page.getByLabel('Green').click();
+  await page.locator('#qty').fill('3');
```
Comment (adrianmaciuc) [inline tests/cart/addItemToCart.spec.js:37]: create a reusable method for this part

**Review by adrianmaciuc — CHANGES_REQUESTED** [review]

File: tests/cart/addItemToCart.spec.js (hunk 3 above)
Comment (adrianmaciuc) [reply tests/cart/addItemToCart.spec.js:37]: also getByTestId('qty')

embrace the builtin methods, live the builtin methods, breath the builtin methods

File: tests/cart/addItemToCart.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import { test, expect } from '@playwright/test';
+const HomePageUrl = 'https://magento.softwaretestingboard.com/';
+
+test('Navigate to top tees', async ({ page }) => {
+  await page.goto(HomePageUrl);
+  await page.locator('#ui-id-5').hover();
+  await page.locator('#ui-id-17').hover();
+  await page.locator('#ui-id-21').click();
```
Comment (adrianmaciuc) [inline tests/cart/addItemToCart.spec.js:8]: getByTestId('ui-id05') then the same for the rest

Comment (pmirel) [reply tests/cart/addItemToCart.spec.js:8]: sounds great, doesn't work
2023-12-12 08_32_56-Playwright Test

File: tests/cart/addItemToCart.spec.js (hunk 3 above)
Comment (pmirel) [reply tests/cart/addItemToCart.spec.js:37]: That is the frst method I tried, but it doesn't find the element

File: tests/homePage/homePageHeader2.spec.js (hunk 2 above)
Comment (pmirel) [reply tests/homePage/homePageHeader2.spec.js:29]: this is from another file, and it's a different test. Are you sure you want me to merge the 2 deifferent tests?

**Review by adrianmaciuc — APPROVED** [review]

**Review by simonleventeattila — APPROVED** [review]

**Pull request merged**